gdrive_sync/api.py

```diff
@@ -557,18 +558,127 @@ def update_sync_status(website: Website, sync_datetime: datetime):
     website.save()
 
 
+def __iter_all_config_items(website: Website) -> Iterator[Tuple[bool, ConfigItem]]:
+    """
+    Yields ConfigItem for all active starters.
+
+    Args:
+        website (Website): The website being scanned.
+
+    Yields:
+        Iterator[Tuple[bool, ConfigItem]]: A generator where yield[0] indicates whether or not the ConfigItem
+            yield[1] belongs to the starter of `website`.
+    """
+    # Assume the starter of websites without a starter to be the course starter.
+    website_starter_slug = (
+        website.starter.slug if website.starter else settings.OCW_COURSE_STARTER_SLUG
+    )
+    all_starters = WebsiteStarter.objects.filter(
+        status__in=WebsiteStarterStatus.ALLOWED_STATUSES
+    )
+
+    for starter in all_starters:
+        for item in SiteConfig(starter.config).iter_items():
+            yield website_starter_slug == starter.slug, item
+
+
+def __iter_related_fields(
+    item: ConfigItem, only_cross_site: bool
+) -> Iterator[ConfigField]:
+    """
+    Yields ConfigField for each field in `item`.
+
+    Args:
+        item (ConfigItem): The ConfigItem being scanned.
+        only_cross_site (bool): Whether or not to yield only cross site fields.
+
+    Yields:
+        Iterator[ConfigField]: A generator that yields ConfigField.
+    """
+    for field in item.fields:
+        if not only_cross_site or field.get("cross_site", False):
+            yield ConfigField(field)
+
+
+def __get_field_filter(field: dict, resource_id: str, website: Website) -> Optional[Q]:
```

Comment:
Again, I think this could be a function directly on the `ConfigField` object.

Reply:
For this, I think it would make more sense for this function to be in a `utils` module. This function only returns for fields that associate with `resource` in some way. The implementation is quite specific too. It will be returning None for the majority of the fields. Additionally, I don't think this is something we will be using quite often.